core/templates/dev/head/domain/exploration/EditableExplorationBackendApiService.ts

```diff
@@ -18,108 +18,123 @@
 
 require('domain/exploration/ReadOnlyExplorationBackendApiService.ts');
 require('domain/utilities/UrlInterpolationService.ts');
+require(
+  'pages/exploration-editor-page/services/exploration-rights-data.service.ts');
 
 require('pages/exploration-player-page/exploration-player-page.constants.ts');
 
 var oppia = require('AppInit.ts').module;
 
 oppia.factory('EditableExplorationBackendApiService', [
-  '$http', '$q', 'ReadOnlyExplorationBackendApiService',
-  'UrlInterpolationService',
+  '$http', '$q', 'ExplorationRightsDataService',
+  'ReadOnlyExplorationBackendApiService', 'UrlInterpolationService',
   'EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE',
   'EDITABLE_EXPLORATION_DATA_URL_TEMPLATE',
   'EXPLORATION_DATA_URL_TEMPLATE',
   'VOICEOVER_EXPLORATION_DATA_URL_TEMPLATE',
-  function($http, $q, ReadOnlyExplorationBackendApiService,
-      UrlInterpolationService,
+  function($http, $q, ExplorationRightsDataService,
+      ReadOnlyExplorationBackendApiService, UrlInterpolationService,
       EDITABLE_EXPLORATION_DATA_DRAFT_URL_TEMPLATE,
       EDITABLE_EXPLORATION_DATA_URL_TEMPLATE,
       EXPLORATION_DATA_URL_TEMPLATE,
       VOICEOVER_EXPLORATION_DATA_URL_TEMPLATE) {
     var _fetchExploration = function(
         explorationId, applyDraft, successCallback, errorCallback) {
-      var editableExplorationDataUrl = _getExplorationUrl(
-        explorationId, applyDraft);
-
-      $http.get(editableExplorationDataUrl).then(function(response) {
-        var exploration = angular.copy(response.data);
-        if (successCallback) {
-          successCallback(exploration);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
+      _getExplorationUrl(explorationId, applyDraft).then(
```

Comment:
Why `_getExplorationUrl` is async? (I  think this can be generated easily, right?)